chains/solana/contracts/programs/ccip-router/src/instructions/v1/onramp.rs

```diff
@@ -307,24 +305,87 @@ fn token_transfer(
 /////////////
 // Helpers //
 /////////////
-fn extra_args_or_default(default_config: Ref<Config>, extra_args: ExtraArgsInput) -> AnyExtraArgs {
-    let mut result_args = AnyExtraArgs {
-        gas_limit: default_config.default_gas_limit.to_owned(),
-        allow_out_of_order_execution: default_config.default_allow_out_of_order_execution != 0,
-    };
 
-    if let Some(gas_limit) = extra_args.gas_limit {
-        gas_limit.clone_into(&mut result_args.gas_limit)
+// process_extra_args returns serialized extraArgs, gas_limit, allow_out_of_order_execution
+// it calls the chain-specific extra args validation logic
+pub fn process_extra_args(
+    dest_config: &DestChainConfig,
+    extra_args: &[u8],
+    message_contains_tokens: bool,
+) -> Result<(Vec<u8>, u128, bool)> {
+    require_gte!(extra_args.len(), 4, CcipRouterError::InvalidInputs);
+
+    let tag: [u8; 4] = extra_args[..4].try_into().unwrap();
+    let mut data = &extra_args[4..];
+
+    match u32::from_be_bytes(dest_config.chain_family_selector) {
+        CHAIN_FAMILY_SELECTOR_EVM => parse_and_validate_evm_extra_args(dest_config, tag, &mut data),
+        CHAIN_FAMILY_SELECTOR_SVM => {
+            parse_and_validate_svm_extra_args(dest_config, tag, &mut data, message_contains_tokens)
```

Comment:
Are we doing this twice? https://github.com/smartcontractkit/chainlink-ccip/pull/488/files#diff-35c2605216d31b008c6dc84b39a597abf74b4be60ca2d0e43ed443d749b64145R152

Reply:
the switch case is defined twice, yes

one function is for parsing/validating extra args, the other is for validating the receiver address